src/test/util_tests.cpp

```diff
@@ -573,17 +575,43 @@ BOOST_AUTO_TEST_CASE(test_FormatParagraph)
 
 BOOST_AUTO_TEST_CASE(test_FormatSubVersion)
 {
+    int temp = 0;
+    int *ptemp = &temp;
+    std::string arch = (sizeof(ptemp) == 4) ? "32bit" : "64bit";
+
     std::vector<std::string> comments;
     comments.push_back(std::string("comment1"));
     std::vector<std::string> comments2;
     comments2.push_back(std::string("comment1"));
     // Semicolon is discouraged but not forbidden by BIP-0014
     comments2.push_back(
         SanitizeString(std::string("Comment2; .,_?@-; !\"#$%&'()*+/<=>[]\\^`{|}~"), SAFE_CHARS_UA_COMMENT));
-    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99900, std::vector<std::string>()), std::string("/Test:0.9.99/"));
-    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99900, comments), std::string("/Test:0.9.99(comment1)/"));
-    BOOST_CHECK_EQUAL(
-        FormatSubVersion("Test", 99900, comments2), std::string("/Test:0.9.99(comment1; Comment2; .,_?@-; )/"));
+    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99800, {}), std::string("/Test:0.9.98(" + arch + ")/"));
+    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99900, comments), std::string("/Test:0.9.99(comment1; " + arch + ")/"));
+    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99900, comments2),
+        std::string("/Test:0.9.99(comment1; Comment2; .,_?@-; ; " + arch + ")/"));
+
+    excessiveBlockSize = 1000000;
+    excessiveAcceptDepth = 40;
+    settingsToUserAgentString();
+    const char *argv_test[] = {"bitcoind", "-uacomment=comment1", "-uacomment=Comment2", "-uacomment=Comment3"};
+    ParseParameters(4, (char **)argv_test, AllowedArgs::Bitcoind());
+    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99900, BUComments),
+        std::string("/Test:0.9.99(EB1; AD40; " + arch + "; comment1; Comment2; Comment3)/"));
+
+    const char *argv_test2[] = {"bitcoind", "-uacomment=Commenttttttttttttttttttttttttttttttttttttttttt1",
+        "-uacomment=Commenttttttttttttttttttttttttttttttttttttttttttttttttttttt2",
+        "-uacomment=Commenttttttttttttttttttttttttttttttttttttttttttttttttttttt3",
+        "-uacomment=Commenttttttttttttttttttttttttttttttttttttttttttttttttttttt4"};
+    ParseParameters(5, (char **)argv_test2, AllowedArgs::Bitcoind());
+    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99900, BUComments),
+        std::string("/Test:0.9.99(EB1; AD40; " + arch + "; Commenttttttttttttttttttttttttttttttttttttttttt1; "
+                                                        "Commenttttttttttttttttttttttttttttttttttttttttttttttttttttt2; "
+                                                        "Commenttttttttttttttttttttttttttttttttttttttttttttttttttttt3; "
+                                                        "Commenttttttttttttttttttttttttttttttttttttttttttt)/"));
+
+    std::string subver = FormatSubVersion("Test", 99900, BUComments);
+    BOOST_CHECK_EQUAL(subver.size(), MAX_SUBVERSION_LENGTH);
 }
```

Comment:
Looking at the excessive blocks tests, I'm wondering if we should set EB, AD, and BUComments back to the defaults at the end of the test, similar to what the buip005 test case does.  Not sure if it is necessary, but since it is affecting global state could potentially cause issues with other tests run after.

Reply:
Agreed, done.